src/plugins/outputPlayedSong/outputPlayedSong.cpp

```diff
@@ -15,7 +18,385 @@ constexpr GH3::QbKey IntroArtistInfoText = "intro_artist_info_text";
+		{
+			break;
+		}
+		SongName[i] = 0x0000;
+	}
+
+	for (unsigned int i = 0; i < SongName.length(); i++)
+	{
+		if (SongName[i] == 0x0020 || (i == SongName.length() - 1))
+		{
+			if (bDuringWord) //the word ended
+			{
+				wordCounter++;
+				bool bAAnTheFound = false;
+				for (int j = 0; j < 3; j++)
+				{
+					if (AAnThe[j].length() == i - LastWordStart + 1 + ((i == SongName.length() - 1) ? (0) : (-1)))
+					{
+						if (wcsncmp(AAnThe[j].c_str(), SongName.substr(LastWordStart, AAnThe[j].length()).c_str(), AAnThe[j].length()) == 0)
+						{
+							if (wordCounter == 0)
+							{
+								wordCounter--; //with "a", "an" or "the" as first word, the following word should be treated as first (recursively) eg. "the a Banana" 
+								bCapitalize = (i == SongName.length() - 1); //if it is the only word in song name (excluding a, an and the) (it also produces "the A" and "a An")
+							}
+							else
+							{
+								bCapitalize = false;
+							}
+							bAAnTheFound = true;
+							break;
+						}
+					}
+				}
+				if (!bAAnTheFound)
+				{
+					for (unsigned int j = 0; j < 19; j++)
+					{
+						if (OtherLowercaseWords[j].length()== i - LastWordStart + 1 + ((i == SongName.length() - 1) ? (0) : (-1)))
+						{
+							if (wcsncmp(OtherLowercaseWords[j].c_str(), SongName.substr(LastWordStart, OtherLowercaseWords[j].length()).c_str(), OtherLowercaseWords[j].length()) == 0)
+							{
+								bCapitalize = wordCounter == 0 || (i == SongName.length() - 1); //only first or last word
+								break;
+							}
+						}
+					}
+				}
+				bDuringWord = false;
+				ApplyWordLowercasing(SongName, LastWordStart, i - LastWordStart + ((i == SongName.length() - 1) ? (0) : (-1)), bCapitalize);
+			}
+		}
+		else
+		{
+			if (!bDuringWord)
+			{
+				bDuringWord = true;
+				LastWordStart = i;
+				bCapitalize = true;//capitalize words by default
+			}
+		}
+	}
+}
+
+std::wstring GetPrefix()
+{
+	//read prefix in
+
+	if (prefSet)
+	{
+		return prefix;
+	}
+
+	prefSet = true;
+
+	std::ifstream aif("prefixansi.txt", std::ios::binary);
+	if (aif.is_open())
+	{
+		std::string sline;
+		std::getline(aif, sline);
+		aif.close();
+		std::wstring w(sline.begin(), sline.end());
+		if (w != L"")
+		{
+			prefix = w;
+			return w;
+		}
+	}
+
+
+	std::wifstream wif("prefixutf16.txt", std::ios::binary);
+	if (wif.is_open())
+	{
+		std::wstring wline;
+		// cut BOM of utf16
+		wif.imbue(std::locale(wif.getloc(), new std::codecvt_utf16<wchar_t, 0x10ffff, std::consume_header>));
+		std::getline(wif, wline);
+		wif.close();
+		prefix = wline;
+		return wline;
+	}
+	prefix = L"";
+	return L"";
+}
+
+std::wstring ProcessFullSongName(std::wstring & Song, std::wstring & Artist, bool AddPrefix = true)
+{
+	std::wstring temp = L"";
+	if (AddPrefix)
+	{
+		temp = prefix;
+	}
+	ProcessUnicodeSongName(Song);
+	ProcessUnicodeArtistName(Artist);
+	temp += Artist + L"-" + Song;
+	return temp;
+}
+
+DWORD WINAPI ProcessSongFileWrite(LPVOID l);
```

Comment:
Delegating this to a thread is unnecessary.  The file write doesn't take long and is only called once per song at the start.  It would be better to focus on optimising the detour to reduce any general stuttering during the song